mypy/types_utils.py

```diff
@@ -81,14 +82,14 @@ def is_bad_type_type_item(item: Type) -> bool:
     Such types are explicitly prohibited by PEP 484. Also, they cause problems
     with recursive types like T = Type[T], because internal representation of
     TypeType item is normalized (i.e. always a proper type).
+
+    Also forbids `Type[Literal[...]]`, because typing spec does not allow it.
```

Comment:
I think it might be better to frame this in terms of what *is* allowed. Here are some other things mypy allows that I don't think make sense: https://mypy-play.net/?mypy=latest&python=3.12&gist=aa900cb6c86fd55901b21ee182aa35ec

Reply:
I suspected that this will be brought up :)
I would prefer to keep this PR focused on just one thing. Because I only agree with the first example: `type[TypeGuard[int]]`, but it is a completely different thing to fix.

Reply:
I think that we can clarify this in https://github.com/python/typing/tree/main/docs/spec

Reply:
This is already specified in the grammar at https://typing.readthedocs.io/en/latest/spec/annotations.html#type-and-annotation-expressions . (Though I noticed we're missing unions there, which should be supported.)

Reply:
I'm fine with deferring other similar changes to other PRs, though.